File: Python/SQLAlchemy_ORM/dal/entities/order.py
from sqlalchemy import Column, DateTime, Float, ForeignKey, Integer, String
from sqlalchemy.orm import relationship
from dal.data.base import Base

class Order(Base):
    __tablename__ = 'Orders'

    OrderID = Column(Integer, primary_key=True)
    CustomerID = Column(String(5), ForeignKey('Customers.CustomerID'))
    EmployeeID = Column(Integer, ForeignKey('Employees.EmployeeID'))
    OrderDate = Column(DateTime)
    RequiredDate = Column(DateTime)
    ShippedDate = Column(DateTime)
    ShipVia = Column(Integer, ForeignKey('Shippers.ShipperID'))
    Freight = Column(Float)
    ShipName = Column(String(40))
    ShipAddress = Column(String(60))
    ShipCity = Column(String(15))
    ShipRegion = Column(String(15))
    ShipPostalCode = Column(String(10))
    ShipCountry = Column(String(15))

    customer = relationship('Customer', back_populates='orders')
    employee = relationship('Employee', back_populates='orders')
    shipper = relationship('Shipper', back_populates='orders')
    order_details = relationship('OrderDetail', back_populates='order')

File: Go/GORM_ORM/dal/entities/order.go
package entities

import "time"

type Order struct {
	OrderID        int           `gorm:"primaryKey;autoIncrement;column:OrderID"`
	CustomerID     *string       `gorm:"size:5;column:CustomerID"`
	EmployeeID     *int          `gorm:"column:EmployeeID"`
	OrderDate      *time.Time    `gorm:"column:OrderDate"`
	RequiredDate   *time.Time    `gorm:"column:RequiredDate"`
	ShippedDate    *time.Time    `gorm:"column:ShippedDate"`
	ShipVia        *int          `gorm:"column:ShipVia"`
	Freight        *float64      `gorm:"type:money;default:0;column:Freight"`
	ShipName       *string       `gorm:"size:40;column:ShipName"`
	ShipAddress    *string       `gorm:"size:60;column:ShipAddress"`
	ShipCity       *string       `gorm:"size:15;column:ShipCity"`
	ShipRegion     *string       `gorm:"size:15;column:ShipRegion"`
	ShipPostalCode *string       `gorm:"size:10;column:ShipPostalCode"`
	ShipCountry    *string       `gorm:"size:15;column:ShipCountry"`
	OrderDetails   []OrderDetail `gorm:"foreignKey:OrderID"`
	Customer       *Customer     `gorm:"foreignKey:CustomerID"`
	Employee       *Employee     `gorm:"foreignKey:EmployeeID"`
	Shipper        *Shipper      `gorm:"foreignKey:ShipVia"`
}
